File: Sqrt(x).java
class Solution {
    public int mySqrt(int x) {
        //min runtime(1ms) S(C(n))=O(1) and (T(C)=(O(log(n))) as it requires non additional space 
        if(x==0 || x==1)return x;//initializing x val to either 0 or 1 and printing it
        int start=1;int mid=-1;int end=x;//start ,mid,end val declare
        while(start<=end){//iterating start to end val 
            mid=start+(end-start)/2;//calculatings qare root val
            if(mid*(long)mid> (long)x)end=mid-1;//checking mid >x and declaring end val to mid's less val
            else if (mid*mid==x)return mid;//x to mid's square assigiment and priniting it 
            else start=mid+1;//incrementing start to mid 
        }return Math.round(end);}}//prinitng the estimated end's val    

File: sqrt(x).cpp
#include<cmath>//including cmath liberary
class Solution {
public:
    int mySqrt(int x) {
        //min.Runtime(0ms.) T(C(N)=O(log(n)) and S(C(N)=O(1)) as it requires non additional space
        if(x==0 || x==1)return x;//Initialize x to 0 or 1 and print x
        int start=1;int mid=-1;int end=x;//varibles declare
        while(start<=end){//iterating x from its start till end 
            mid=start+(end-start)/2;//calcualting sqaure root val
            if(mid*(long)mid >(long)x)end=mid-1;//decrementing the val from square of its num
            else if(mid*mid==x)return mid;//printing the mid val to its square val
            else start=mid+1;//incrementing the mid val
        }return round(end);}};//printing the estimated val
